CRM/Core/Form/RecurringEntity.php

```diff
@@ -399,6 +399,7 @@ public static function postProcess($params = array(), $type, $linkedEntities = a
                 'is_active' => 1,
               );
               $excludeDateList[] = $optionGroupValue['value'];
+              // Something is wrong here. The returned BAO object is not being used.
               CRM_Core_BAO_OptionValue::create($optionGroupValue);
```

Comment:
Yes I was a bit confused regarding the issue at that point of time. However I recreated the issue on my machine and hopefully now I'm on the right track.
My take on solving the issue is to add a query "SET NAMES utf8" to CRM_Admin_Forms_Options::preProcess function. I'm still not sure where the query fits in exactly though.